Configuration/Applications/python/ConfigBuilder.py

```diff
@@ -2268,19 +2268,31 @@ def prepare(self, doChecking = False):
 
-        if self._options.nThreads != "1":
+        overrideThreads = (self._options.nThreads != "1")
+        overrideConcurrentLumis = (self._options.nConcurrentLumis != defaultOptions.nConcurrentLumis)
+        overrideConcurrentIOVs = (self._options.nConcurrentIOVs != defaultOptions.nConcurrentIOVs)
+
+        if overrideThreads or overrideConcurrentLumis or overrideConcurrentIOVs:
             self.pythonCfgCode +="\n"
             self.pythonCfgCode +="#Setup FWK for multithreaded\n"
-            self.pythonCfgCode +="process.options.numberOfThreads = "+self._options.nThreads+"\n"
-            self.pythonCfgCode +="process.options.numberOfStreams = "+self._options.nStreams+"\n"
-            self.pythonCfgCode +="process.options.numberOfConcurrentLuminosityBlocks = "+self._options.nConcurrentLumis+"\n"
-            self.pythonCfgCode +="process.options.eventSetup.numberOfConcurrentIOVs = "+self._options.nConcurrentIOVs+"\n"
-            if int(self._options.nConcurrentLumis) > 1:
-              self.pythonCfgCode +="if hasattr(process, 'DQMStore'): process.DQMStore.assertLegacySafe=cms.untracked.bool(False)\n"
-            self.process.options.numberOfThreads = int(self._options.nThreads)
-            self.process.options.numberOfStreams = int(self._options.nStreams)
-            self.process.options.numberOfConcurrentLuminosityBlocks = int(self._options.nConcurrentLumis)
-            self.process.options.eventSetup.numberOfConcurrentIOVs = int(self._options.nConcurrentIOVs)
+            if overrideThreads:
+                self.pythonCfgCode +="process.options.numberOfThreads = "+self._options.nThreads+"\n"
+                self.pythonCfgCode +="process.options.numberOfStreams = "+self._options.nStreams+"\n"
+                self.process.options.numberOfThreads = int(self._options.nThreads)
+                self.process.options.numberOfStreams = int(self._options.nStreams)
+            if overrideConcurrentLumis:
+                self.pythonCfgCode +="process.options.numberOfConcurrentLuminosityBlocks = "+self._options.nConcurrentLumis+"\n"
+                self.process.options.numberOfConcurrentLuminosityBlocks = int(self._options.nConcurrentLumis)
+            if overrideConcurrentIOVs:
+                self.pythonCfgCode +="process.options.eventSetup.numberOfConcurrentIOVs = "+self._options.nConcurrentIOVs+"\n"
+                self.process.options.eventSetup.numberOfConcurrentIOVs = int(self._options.nConcurrentIOVs)
+        if self._options.nConcurrentLumis != "1":
+            # nConcurrentLumis == 0 implies that framework decides the
+            # value based on the number of streams, and the number of
+            # threads/streams can be set afterwards. The safest check
+            # is then to set assertLegacySafe == True only when
+            # explicitly asked to use 1 concurrent lumi.
+            self.pythonCfgCode +="if hasattr(process, 'DQMStore'): process.DQMStore.assertLegacySafe=cms.untracked.bool(False)\n"
```

Comment:
@cms-sw/dqm-l2 I started to wonder if we could just set this parameter to `True` by default? (and eventually remove it as part of removal of support for legacy modules)

Reply:
Alternatively DQMStore could use `watchPreallocate` signal to check if the job has at least two concurrent lumis to disable this assert.

Reply:
@makortel we support preventing concurrent lumis with the changes mentioned above until we complete the migration from legacy modules (and legacy way to access DQMStore). Either method is fine for us, thought it looks like the first one is simpler, so perhaps preferred.

Reply:
Sorry, I'm mixing things. The `assertLegacySafe` was set to `False` (which is what I should have written in the first message instead of `True`) by default already in https://github.com/cms-sw/cmssw/pull/34366. So this setting doesn't do anything anymore, and can be removed.